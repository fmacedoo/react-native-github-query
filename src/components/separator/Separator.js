import React from 'react';

import { StyledView } from './styles';

const Separator = () => <StyledView testID="Separator" />;

export default Separator;
