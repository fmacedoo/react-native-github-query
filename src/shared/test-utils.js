import { ThemeProvider } from 'styled-components/native';
import { render } from '@testing-library/react-native';
import React from 'react';

import { gitHubTheme as theme } from '@reactnativegithubquery/themes';

const AllTheProviders = ({ children }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render, theme };
