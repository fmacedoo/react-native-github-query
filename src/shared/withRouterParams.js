import React from 'react';

export default function withRouterParams(WrappedComponent) {
    return ({ route: { params } }) => <WrappedComponent {...params} />;
}
