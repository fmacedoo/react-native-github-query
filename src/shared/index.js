import withRouterParams from './withRouterParams';

export { withRouterParams };
