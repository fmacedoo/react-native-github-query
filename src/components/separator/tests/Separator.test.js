import 'jest-styled-components';
import 'react-native';
import React from 'react';

import Separator from '../Separator';

import { render } from '@reactnativegithubquery/shared/test-utils';

describe('<Separator /> rendering', () => {
    it('Separator with default props', () => {
        const tree = render(<Separator />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
