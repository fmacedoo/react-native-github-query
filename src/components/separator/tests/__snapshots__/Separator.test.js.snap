// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator /> rendering Separator with default props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#768390",
        "height": 0.5,
        "width": "100%",
      },
    ]
  }
  testID="Separator"
/>
`;
